evm/p2p/discovery.py

```diff
@@ -276,7 +276,7 @@ def _unpack(message: AnyStr) -> Tuple[datatypes.PublicKey, int, List[Any], AnySt
     return remote_pubkey, cmd_id, payload, message_hash
 
 
-if __name__ == "__main__":
+def main():
```

Comment:
The actual problem being fixed here is that all variables declared in the "main function" (before I made it a function) become global variables and are reachable by all functions being called from it. The actual warnings from pylint was that every single variable declared didn't conform to UPPER_CASE naming style (because they're global), and that many other random functions that used the same variable names got "Redefining name <name> from outer scope".
If we want to indicate that the main function shouldn't be called I can name it something less inviting, but if we really don't want the code in a function the variables would need to be named __PRIVKEY or similar to hope that other functions don't use them.
Could move the main function into a separate file such that you'd have to explicitly import it (and place a comment in it that it's not to be imported and used) and don't automatically get it when importing discovery.py

Reply:
Well, all of this is only a problem if we intend to run pylint as part of our CI checks, right? Is there actually much value in running pylint as well as flake8?

Reply:
Isn't it a problem that the variables are global? It would e.g. make it easy to forget to define the variable yourself in an underlying function and instead accidentally use the global variable.
I picked up pylint mostly for the fun of it, so getting rid of specific warnings is probably not a priority atm and I will not bother adding pylint's version of noqa for now.
Can probably evaluate later whether the stuff that I find with pylint is enough to warrant integrating it into the CI or should just do it as a one-off.

Reply:
Yeah, that could be a problem and it doesn't look like any of our CI checks would prevent that. Maybe just using a name like `_test()` for those functions would be enough?